kedro-telemetry/pyproject.toml

```diff
@@ -43,10 +43,10 @@ test = [
 ]
 
 [project.entry-points."kedro.cli_hooks"]
-kedro-telemetry = "kedro_telemetry.plugin:cli_hooks"
+kedro-telemetry = "kedro_telemetry.plugin:telemetry_hook"
 
 [project.entry-points."kedro.hooks"]
-kedro-telemetry = "kedro_telemetry.plugin:project_hooks"
+kedro-telemetry = "kedro_telemetry.plugin:telemetry_hook"
```

Comment:
It looks weird to me that these are now exactly the same.. Can you explain a bit more why it's necessary to combine the hooks?

Reply:
The conversation can be found here: https://kedro-org.slack.com/archives/C054U0LJLAC/p1720714959292789

In short. Kedro has two hooks manager, one for cli, the other for the "usual" one. If the hook is set in `settings.py` instead of registered through entrypoint, you will find that only the `@hook_impl` method is invoked but not the `@cli_hook_impl`.

I tried to argue it may make sense to combine these hooks into one. In theory it's nice to have the hook manager separated, but when it comes to actual usage, you need to implement hooks for both and starting to get diverge behavior unless you know that they are handled by two different hook manager(internal details). Or, alternatively, we  keep one `@hook_impl` but each hook manager handle a subset of hookspec (i.e. before_command_run).

^ This is a discussion for longer-term 

You can see that they are handled separately in Kedro

For regular hook entrypoint / settings.py: https://github.com/kedro-org/kedro/blob/1aea4a3d6218ba50c0ae89945bf2b38d751fbfca/kedro/framework/session/session.py#L117-L120

For cli hook: https://github.com/kedro-org/kedro/blob/1aea4a3d6218ba50c0ae89945bf2b38d751fbfca/kedro/framework/cli/hooks/manager.py#L16

Reply:
cli_hook: https://github.com/kedro-org/kedro/blob/1aea4a3d6218ba50c0ae89945bf2b38d751fbfca/kedro/framework/cli/hooks/manager.py#L37

hook: https://github.com/kedro-org/kedro/blob/1aea4a3d6218ba50c0ae89945bf2b38d751fbfca/kedro/framework/hooks/manager.py#L68

This is the direct reason why it has to be defined twice in `pyproject.toml`, essentially the same object is being passed to two different entrypoint, and then the corresponding hook manager execute it accordingly.

Reply:
I tried to argue it may make sense to combine these hooks into one. In theory it's nice to have the hook manager separated, but when it comes to actual usage, you need to implement hooks for both and starting to get diverge behavior unless you know that they are handled by two different hook manager(internal details). Or, alternatively, we  keep one `@hook_impl` but each hook manager handle a subset of hookspec (i.e. before_command_run).

^ This is a discussion for longer-term 

I would like to simplify this at least in the "front end" where user doesn't need to know the different between these two entrypoints. IMO it's redundant as well, as we can already different CLI specific entrypoint via `@cli_hook_impl` vs `@hook_impl`. Or we can even just tell the different from the name of the method.

Reply:
I read the Slack conversation beforehand, but I was confused by the implementation because it seemed to do the opposite of what was said in Slack? 

> @idanov said: "it doesn't really sit right - single responsibility principle, meaning CLI hooks and Runtime events are better suited to be separate (at least to my eyes) "